results/cmd/watcher/main.go

```diff
@@ -100,20 +108,76 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
 		return err
 	}
 
-	r.logger.Infof("Sending update for %s/%s (uid %s)", namespace, tr.Name, tr.UID)
+	r.logger.Infof("Recieving new TaskRun %s/%s", namespace, tr.Name)
 
 	// Send the new status of the TaskRun to the API server.
 	p, err := convert.ToProto(tr)
 	if err != nil {
 		r.logger.Errorf("Error converting to proto: %v", err)
 		return err
 	}
-	if _, err := r.client.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{
-		TaskRun: p,
-	}); err != nil {
-		r.logger.Error("Error updating TaskRun %s: %v", name, err)
-		return err
-	}
 
+	// Create a TaskRun if it does not exist in results server, update existing one otherwise.
+	annotation := p.GetMetadata().GetAnnotations()
```

Comment:
nit: We only ever use this once, so feel free to inline below.

Reply:
done